beets/library.py

```diff
@@ -102,8 +101,11 @@ def col_clause(self):
         escape = lambda m: self.escape_char + m.group(0)
         dir_pattern = self.escape_re.sub(escape, self.dir_path)
         dir_blob = buffer(dir_pattern + b'%')
-        return '({0} = ?) || ({0} LIKE ? ESCAPE ?)'.format(self.field), \
-               (file_blob, dir_blob, self.escape_char)
+        file_pattern = self.escape_re.sub(escape, self.file_path)
+        file_blob = buffer(file_pattern + b'%')
```

Comment:
I think we don't want the `+ b'%'` here, which will let the query `path:/my/song.mp3` match a file at path `/my/song.mp3.foo`. The `+ b'%'` is only needed for the directory part of the query (`dir_pattern`).